libparsec/crates/client/src/workspace/merge.rs

```diff
@@ -23,35 +23,122 @@ pub(super) enum MergeLocalFolderManifestOutcome {
+    // containing only zero-filled data between bytes 1024 and 2048:
+    //
+    //          0                1024               2048               3072
+    // remote   |<----block #1---->|                  |<----block #2---->|
+    // local    |<-non-empty area->|<---empty area--->|<-non-empty area->|
+    //
+    // Here the two non-empty areas in local are expected to each contains a single
+    // chunk view corresponding to the same block than in remote.
+
+    // Compare each blocksize area one by one.
+    let mut remote_blocks_iter = base_blocks.iter();
+    for local_blocksize_area in local_blocks.iter() {
+        match local_blocksize_area.len() {
+            // This blocksize area is empty (i.e. it contains only zero-filled data), the
+            // remote manifest should have a hole here (hence nothing to compare here).
+            0 => (),
+            // This blocksize area contains data to compare.
+            1 => {
+                // The blocksize area is made of a single chunk view in local, from there
+                // we can extract the corresponding block access and compare it to the
+                // remote one.
+                // Note that it's possible to have a chunk view referring a block access
+                // but only using part of it (think of truncating a synchronized file),
+                // in this case `ChunkView::get_block_access` works as expected and
+                // return an error (given the chunk view doesn't *correspond* to the block
+                // access but only *uses* it).
+                match (local_blocksize_area[0].get_block_access(), remote_blocks_iter.next()) {
+                    // The local manifest contains local changes in this blocksize area.
+                    (Err(ChunkViewGetBlockAccessError::NotPromotedAsBlock), _)
+                    // The local manifest contains more blocksize areas than the remote.
+                    | (_, None) => {
+                        return true;
+                    }
+                    (Ok(local_block_access), Some(remote_block_access)) => {
+                        // This blocksize area has been modified.
+                        if local_block_access != remote_block_access {
+                            return true;
+                        }
+                    }
+                }
+            }
+            // Multiple chunk views in a given blocksize area indicates there is local
+            // changes (i.e. reshape hasn't been done).
+            _ => {
+                return true;
+            }
+        }
+    }
+
+    // No changes \o/
+    false
+}
+
 pub(super) fn merge_local_file_manifest(
     local_author: DeviceID,
     timestamp: DateTime,
-    local_manifest: &LocalFileManifest,
-    remote_manifest: FileManifest,
+    local: &LocalFileManifest,
+    remote: FileManifest,
 ) -> MergeLocalFileManifestOutcome {
     // 0) Sanity checks, caller is responsible to handle them properly !
```

Comment:
It would be nice to add that to the function doc

Reply:
I've added a docstring similar to the one for `merge_local_folder_manifest`